deezer_downloader/deezer.py

```diff
@@ -392,20 +395,99 @@ def download_song(song: dict, output_file: str) -> None:
             raise
 
     key = calcbfkey(song["SNG_ID"])
+    is_flac = get_file_extension() == "flac"
+
     try:
         with session.get(url, stream=True) as response:
             response.raise_for_status()
             with open(output_file, "w+b") as fo:
-                # Add song cover and first 30 seconds of unencrypted data
-                writeid3v2(fo, song)
-                decryptfile(response, key, fo)
-                writeid3v1_1(fo, song)
+                if is_flac:
+                    # For FLAC files, we'll add metadata after downloading
+                    decryptfile(response, key, fo)
+                else:
+                    # For MP3 files, add ID3 tags as before
+                    writeid3v2(fo, song)
+                    decryptfile(response, key, fo)
+                    writeid3v1_1(fo, song)
+
+        # Add metadata to FLAC files using mutagen
+        if is_flac:
+            add_flac_metadata(output_file, song)
```

Comment:
let's have a function add_metadata(). That function should differentiate between mp3 and flac